js/PixelPainter.js

```diff
@@ -0,0 +1,386 @@
+
+
+
+function makeColorPalette(height, width = height) {
+
+  // keeping height as parameter in case i want to make a gradient
+  function colorStyle(rgbParamNbr, height) {
+    let r = 255;
+    let g = 255;
+    let b = 255;
+
+    // commented code would make a gradient
+    if (rgbParamNbr === 1) {
+      g = Math.floor(Math.random() * 256);
+      b = Math.floor(Math.random() * 256);
+      //   g = Math.floor((255 / height)*(height-1));
+      //   b = Math.floor((255 / height)*(height-1));
+    }
+
+    if (rgbParamNbr === 2) {
+      r = Math.floor(Math.random() * 256);
+      b = Math.floor(Math.random() * 256);
+      //   r = Math.floor((255 / height)*(height-1));
+      //   b = Math.floor((255 / height)*(height-1));
+    }
+
+    if (rgbParamNbr === 3) {
+      r = Math.floor(Math.random() * 256);
+      g = Math.floor(Math.random() * 256);
+      //   r = Math.floor((255 / height)*(height-1));
+      //   g = Math.floor((255 / height)*(height-1));
+    }
+
+    let strRGB = "rgb(" + r + ", " + g + ", " + b + ")";
+
+    return strRGB;
+  }
+
+  const paletteBox = document.createElement('div');
+  paletteBox.id = 'paletteArea';
+  pixelPainter.appendChild(paletteBox);
+  const newElem = document.createElement('div');
+  newElem.id = 'palette';
+  paletteBox.appendChild(newElem);
+  for (let i = 1; i <= height; i++) {
```

Comment:
You could reuse your `makeCanvas` function here if you pass in the element the grid cell should end up being appended too.